### ISSUE TITLE
[v2 FEATURE]: Expanded --save-smiles-splits

### ISSUE THREAD

**Issue by cjmcgill** [chemprop/chemprop#891]
**Is your feature request related to a problem? Please describe.**
The current version of splits output from `--save-smiles-splits` is more limited than in v1. Previously, the output would include a `train_smiles.csv` file that had columns with just the smiles, a `train_full.csv` file that had columns with the targets as well, and a `train_features.csv` that had additional molecular features with the split. Currently, only the `train_smiles.csv` is provided (with integers as the column headers). 

**Use-cases/examples of this new feature**
With the previous level of detail, the `--save-smiles-splits` function generated split files that could then be used directly for prediction jobs of the train, val, or test. Including the features file made that possible for jobs that used features.

By providing the `training_full.csv` version of the file, the user could then readily calculate errors and metrics between them, or provide the file for uncertainty evaluation in those tools.

Also, by providing the features file, users can distinguish datapoints that use the same smiles but have different features as far as where they were split. Such as properties that vary with temperature. Presumably the user would want to use a split that kept repeated smiles together (random_with_repeated_smiles), but in case they didn't this would distinguish between those datapoints.

**Desired solution/workflow**
I would recommend expanding --save-smiles-splits at least back to its previous capability. Maybe we can build retrieval for features and targets into MoleculeDataset class.

**Comment by KnathanM** [comment]
I think the main use case of this proposal is to be able to have `chemprop train` output files that can be used with `chemprop predict` to then see how the model performs on the training set vs the validation set vs the test set. A second use case of this proposal is to save the splits generated by `chemprop` to help users track down which of train/val/test a certain datapoint ended up in. 

For the main case, the main problem is that there are multiple input files to `chemprop`: `--data-path`, `--descriptors-path`, `--atom-features-path`, `--atom-descriptors-path`, and `--bond-features-path`. Atom and bond predictions also can have `--bond-descriptors-path` and `--constraints-path`. We could split each of these files and put them in train, val, and test directories under the `--output-dir`. This would be the easiest to implement. 

Alternatively we could combine all this information into train.csv, val.csv, and test.csv files. In #1250 we are allowing datapoint descriptors to go in the input file. Currently the atom/bond features/descriptors need to be in a `.npz` format. This is because these are a list of arrays (one for each molecule) instead of one array with one row for each molecule. But in v2.2 we added support for atom/bond targets in the input file and atom/bond features/descriptors can follow the same code style. Essentially for a given column, each row contains a string that looks like a list containing the property values for all atoms/bonds in the molecule, ordered the same as they will be in the `Chem.Mol`. (Bond values can also be put in a matrix of shape n_atoms x n_atoms.) This method will take more work to implement but could be useful for some users.

Either way the user will need to write a new command for the prediction loops on each set of data. One downside of both of these methods though is it creates a copy of the input data for each data split. If multiple data splits are used (`--num_replicates` > 1) multiple copies would be made. This could be bad, but would only happen if the user specifies `--save-smiles-splits`. But it is probably good practice to always save the split so the split can be reproduced later. 

After writing all of this, the solution I am leaning towards is to just scrap `--save-smiles-splits` altogether and always save a `splits.json` file to the output dir that follows the format [here](https://chemprop.readthedocs.io/en/latest/tutorial/cli/train.html#train-validation-test-splits). This takes up minimal space and is sufficient for recreating the split. We can also provide a separate script that given a `splits.json` and the input files, splits the input files into train, val, and test directories for integration with `chemprop predict`. 

If there are multiple splits we could either save each split in a separate `splits1.json`, `splits2.json`, etc. or have the script produce split1/train, split1/val, split1/test, split2/train, etc. or train/split1, train/split2, val/split1, etc. All of them seem messy, but I lean towards the last option as it then the output dir won't become cluttered with many files, only three. 

Lastly I don't see the need to have `train_smiles.csv` and `train_full.csv`. I think we should just load the input file using pandas, split it, and save it using pandas so that the output files look the same as the input expect being a subset of the rows.
